### DISCUSSION TITLE
When withoutoverlapping method set in scheduler command never runs

### DISCUSSION THREAD

**Discussion by AliAlican** [laravel/framework#37255] [Q&A]
- Laravel Version: ^8.0
- PHP Version: 7.3.0

### Description:
I want to run a nonoverlapping job but the command is just never called

Here is my code

`$schedule->command('start_listeners')->withoutOverlapping();`
I also tried this

`$schedule->command('start_listeners')->everyMinute()->withoutOverlapping();`
When i remove the withoutOverlapping method it works perfectly fine

![image](https://user-images.githubusercontent.com/44370554/116929703-19397a80-ac5f-11eb-83c9-a43eaee1ec54.png)

**Comment by derMart** [comment]
same problem here

**Comment by Krisell** [comment]
So what happens / what output do you get when running `php artisan schedule:run`?

**Reply by derMart** [reply]
- If the command has `withoutOverlapping()` I get: `No scheduled commands are ready to run.`
- If the command does not have this flag, I get: `[2022-01-10T16:11:48+01:00] Running scheduled command: '/usr/bin/php7.4' 'artisan' command:name > '/dev/null' 2>&1`

**Reply by Krisell** [reply]
I think the overlapping mutex lock uses the cache by default, and the expiration time in case the lock is not released is 24 hours. So if you did run the command but it crashed, the lock might be left and you would need to clear the cache.

**Reply by derMart** [reply]
Thank you, yes in deed running `php artisan cache:clear`fixes my problem.
Still, of course that does not fix that `php artisan schedule:run` will fire commands repeatedly within one minute, see https://github.com/laravel/framework/issues/40325 Please reopen that issue until the behaviour is documented or consider a possible solution: https://github.com/laravel/framework/issues/40325#issuecomment-1008964233

**Reply by derMart** [reply]
> I think the overlapping mutex lock uses the cache by default, and the expiration time in case the lock is not released is 24 hours. So if you did run the command but it crashed, the lock might be left and you would need to clear the cache.

Isn't the lock released when the command crashes, e.g. by encapsulating the command in a try catch block?

**Reply by Krisell** [reply]
Yes that sounds reasonable, and I'm just guessing the cause for you here. Perhaps a `ctrl+c` or similar left it in this state?

**Reply by derMart** [reply]
> Yes that sounds reasonable,

Is it actually encapsulated by laravel at the moment, or isn't it?

There was no user cancellation like ctrl+c, I am pretty sure. There might have been crashes of the command during development, but these should have been caught, if it would have been encapsulated by laravel, but you did not answer that. I am sure you are much faster in finding this out than me (still getting to know laravel).

**Reply by Krisell** [reply]
Yes it is, check in `WithoutOverlapping.php`

![image](https://user-images.githubusercontent.com/25909128/148808068-1d0315d9-96f2-494f-89d2-ea333c31670b.png)

**Reply by derMart** [reply]
Thanks alot! Well then it must have been something else which causes the lock not to get released. Unable to figure this out atm sadly. If it happens again I will try to boil it down.
